dev/tasks/conda-recipes/r-arrow/build.sh

```diff
@@ -1,9 +1,14 @@
-#!/usr/bin/env bash
-
+#!/bin/bash
 set -ex
 
 export DISABLE_AUTOBREW=1
-# See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
-export ARROW_R_CXXFLAGS="${ARROW_R_CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
 
-$R CMD INSTALL --build r/.
+# set C++17 due to abseil
+export ARROW_R_CXXFLAGS="${ARROW_R_CXXFLAGS} -std=c++17"
```

Comment:
I realized later that this comment is not 100% accurate; it's less about abseil, and more about arrow having move the C++17, full stop. Unless people want a specific fix for that, I'll fix that in the feedstock and it'll be part of the next sync.